main_test.go

```diff
@@ -1,62 +1,198 @@
 package main
 
 import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
 	"testing"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestMain(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "main")
+func GetTestConfig() Config {
+
+	originUrl, _ := url.Parse("http://origin.url")
+	proxyUrl, _ := url.Parse("http://proxy.url")
+	ssoUrl, _ := url.Parse("http://sso.url")
+
+	config := Config{
+		OriginURL:       originUrl,
+		ProxyURL:        proxyUrl,
+		SSOURL:          ssoUrl,
+		SSOSecret:       "secret",
+		AllowAll:        false,
+		AllowGroups:     "",
+		BasicAuth:       "",
+		Whitelist:       "",
+		UsernameHeader:  "username-header",
+		GroupsHeader:    "groups-header",
+		Timeout:         10,
+		SRVAbandonAfter: 600,
+		LogRequests:     false,
+	}
+
+	return config
 }
 
-var _ = Describe("parseCookie", func() {
-	var (
-		signed, secret string
+type SSOOptions struct {
+	URL    string
+	Secret string
+	Nonce  string
+	Groups string
+	Admin  bool
+}
 
-		parsedUsername string
-		parsedGroup    string
-		parseError     error
-	)
+func GetSSOOptions(url string, secret string) SSOOptions {
+	options := SSOOptions{
+		URL:    url,
+		Secret: secret,
+		Nonce:  "",
```

Comment:
You may omit [zero values](https://golang.org/ref/spec#The_zero_value), like empty strings and `false`.